File: WinForms/VB/SchedulerMainDemo/Modules/Reminders.vb
Imports DevExpress.DXperience.Demos
Imports System

Namespace DevExpress.XtraScheduler.Demos

    Public Partial Class RemindersModule
        Inherits TutorialControl

        Public Sub New()
            ' This call is required by the Windows Form Designer.
            InitializeComponent()
        ' TODO: Add any initialization after the InitializeComponent call
        End Sub

        Public Overrides ReadOnly Property PrintingSchedulerControl As SchedulerControl
            Get
                Return schedulerControl
            End Get
        End Property

        Private Sub RemindersModule_VisibleChanged(ByVal sender As Object, ByVal e As EventArgs)
            schedulerDataStorage.Appointments.Clear()
            schedulerControl.DataStorage.EnableReminders = Visible
            'schedulerControl.OptionsBehavior.ShowRemindersForm = this.Visible;
            schedulerControl.Start = TutorialConstants.Today
        End Sub

        Private Sub btnCreateAppointmentWithReminder_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim now As Date = TutorialConstants.Now + TimeSpan.FromMinutes(5)
            Dim apt As Appointment = schedulerDataStorage.CreateAppointment(AppointmentType.Normal)
            apt.Start = now
            apt.Duration = TimeSpan.FromHours(1)
            apt.Subject = "Appointment with Reminder"
            apt.HasReminder = True
            schedulerDataStorage.Appointments.Add(apt)
            schedulerControl.ActiveView.GotoTimeInterval(New TimeInterval(apt.Start, apt.Duration))
        End Sub

        Private Sub btnGotoToday_Click(ByVal sender As Object, ByVal e As EventArgs)
            schedulerControl.GoToToday()
        End Sub
    End Class
End Namespace

File: WinForms/CS/SchedulerMainDemo/Modules/Reminders.cs
using DevExpress.DXperience.Demos;
using System;

namespace DevExpress.XtraScheduler.Demos {
    public partial class RemindersModule : DevExpress.XtraScheduler.Demos.TutorialControl {
        public RemindersModule() {
            // This call is required by the Windows Form Designer.
            InitializeComponent();

            // TODO: Add any initialization after the InitializeComponent call
        }
        public override SchedulerControl PrintingSchedulerControl { get { return schedulerControl; } }


        private void RemindersModule_VisibleChanged(object sender, System.EventArgs e) {
            schedulerDataStorage.Appointments.Clear();
            schedulerControl.DataStorage.EnableReminders = this.Visible;
            //schedulerControl.OptionsBehavior.ShowRemindersForm = this.Visible;
            schedulerControl.Start = TutorialConstants.Today;
        }
        private void btnCreateAppointmentWithReminder_Click(object sender, System.EventArgs e) {
            DateTime now = TutorialConstants.Now + TimeSpan.FromMinutes(5);
            Appointment apt = schedulerDataStorage.CreateAppointment(AppointmentType.Normal);
            apt.Start = now;
            apt.Duration = TimeSpan.FromHours(1);
            apt.Subject = "Appointment with Reminder";

            apt.HasReminder = true;
            schedulerDataStorage.Appointments.Add(apt);
            schedulerControl.ActiveView.GotoTimeInterval(new TimeInterval(apt.Start, apt.Duration));
        }

        private void btnGotoToday_Click(object sender, System.EventArgs e) {
            schedulerControl.GoToToday();
        }
    }
}
